### PR TITLE
[ICD] Fix multiple check-in/peer node ids handling in icd client

### LINKED ISSUE #34799: [CERT-TEST-FAILURE] [TC-ICDB-1.3] Test failed - DUT Fails to Refresh Entry with New Key - Receives Status 0x7f (UNSUPPORTED_ENDPOINT) in Step 2b
### Feature Area

Area:ICDPhase2

### Test Case

TC-ICDB-1.3

### Reproduction steps

 At step 2b. According to the test plan, when the DUT receives a Check-In message with a Check-In Counter ICDCounter2 indicating that 2^31 counter values have been used, the DUT should refresh its entry with a new key Key2 via the RegisterClient command to the TH with the following details:
CheckInNodeID: CheckInNodeID
MonitoredSubject: MonitoredSubID
Key: Key2
However, in the current SDK, TH receiving the status 0x7f (UNSUPPORTED_ENDPOINT).

Command used to launch the DUT :./lit-icd-app --enable-key 00112233445566778899AABBCCDDEEFF
Command used to launch the TH(chip-tool) : ./chip-tool pairing onnetwork 1 20202021
After the successful pairing entered to interactive mode using mentioned command -   ./chip-tool interactive start
step 1 : TH sends {C_REGISTER_CLIENT} command.
- CheckInNodeID: CheckInNodeID
- MonitoredSubject: MonitoredSubID
- Key: Key1
  icdmanagement register-client 112233 1 hex:abcdef1234567890abcdef1234567890 0 1 0
> [1722922531.490] [11122:11124] [DMG] Received Command Response Data, Endpoint=0 Cluster=0x0000_0046 Command=0x0000_0001
> [1722922531.490] [11122:11124] [TOO] Endpoint: 0 Cluster: 0x0000_0046 Command 0x0000_0001
> [1722922531.490] [11122:11124] [TOO]   RegisterClientResponse: {
> [1722922531.490] [11122:11124] [TOO]     ICDCounter: 579832728
> [1722922531.490] [11122:11124] [TOO]    }
> [1722922531.491] [11122:11124] [DMG] ICR moving to [AwaitingDe]

step 2a : TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.ICDB.S.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.ICDB.S.TEST_EVENT_TRIGGER for Invalidate ICD half counter values event.
generaldiagnostics test-event-trigger hex:00112233445566778899AABBCCDDEEFF 0x0046000000000003 1 0
> [1722922538.436] [11122:11124] [DMG] Received Command Response Status for Endpoint=0 Cluster=0x0000_0033 Command=0x0000_0000 Status=0x0
> [1722922538.436] [11122:11124] [DMG] ICR moving to [AwaitingDe]

**waited DUT refresh its entry with a new key Key2 via {C_REGISTER_CLIENT} command to TH: - CheckInNodeID: CheckInNodeID
- MonitoredSubject: MonitoredSubID
- Key: Key2**
**However, in the current SDK, TH receiving the status 0x7f (UNSUPPORTED_ENDPOINT).** 

> [1722922559.028] [11122:11124] [DMG] },
> [1722922559.028] [11122:11124] [DMG] No command 0x0000_0000 in Cluster 0x0000_0046 on Endpoint 0x0
> [1722922559.028] [11122:11124] [DMG] Endpoint=0 Cluster=0x0000_0046 Command=0x0000_0000 status 0x7f (UNSUPPORTED_ENDPOINT) (no additional context)
> [1722922559.028] [11122:11124] [DMG] Command handler moving to [NewRespons]

### Bug prevalence

Everytime

### GitHub hash of the SDK that was being used

b9c192b7cf8e7ffc697fa9b25eaeb6c5faa09365

### Platform

raspi

### Anything else?

**PFA Log below** 
[TC-ICDB-1.3.txt](https://github.com/user-attachments/files/16505657/TC-ICDB-1.3.txt)
[TC-ICDB-1.3 DUT.txt](https://github.com/user-attachments/files/16505658/TC-ICDB-1.3.DUT.txt)

**Test plan reference** 
https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/icdbehavior.adoc#tc-icdb-1-3-icd-check-in-protocol-client-response-dut_client

### PR DESCRIPTION
**Pull request by yunhanw-google**
When running icd registration and refresh key, we have problematically store check-in nodeId in peer nodeId. In order to resolve this issue, we add a new check_in_node in icdClientInfo, and update the code from bottom to up across android/python/c++ chiptool.

Tests:
Validate using chip-tool
./out/linux-x64-chip-tool/chip-tool interactive start
pairing onnetwork-long  1 20202021 3840
icdmanagement register-client 112233 1 hex:abcdef1234567890abcdef1234567890 0 1 0
icd wait-for-device 1 30

pairing onnetwork-long  1 20202021 3840 --icd-registration=true
icd wait-for-device 1 30

fixes: https://github.com/project-chip/connectedhomeip/issues/34799

### CODE REVIEW

**Review by joonhaengHeo — APPROVED** [review]

File: examples/chip-tool/commands/clusters/ClusterCommand.h [hunk 1]
```diff
@@ -159,7 +163,7 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
         if ((path.mEndpointId == chip::kRootEndpointId) && (path.mClusterId == chip::app::Clusters::IcdManagement::Id) &&
             (path.mCommandId == chip::app::Clusters::IcdManagement::Commands::UnregisterClient::Id))
         {
-            ClearICDEntry(mScopedNodeId);
+            ClearICDEntry(mPeerNodeId);
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ClusterCommand.h:166]: OK, so UnregisterClient does not use `mCheckInNodeId`?  Why are we setting it there, then?

File: examples/chip-tool/commands/clusters/ClusterCommand.h [hunk 2]
```diff
@@ -260,7 +264,8 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
 private:
     chip::ClusterId mClusterId;
     chip::CommandId mCommandId;
-    chip::ScopedNodeId mScopedNodeId;
+    chip::ScopedNodeId mPeerNodeId;
+    chip::ScopedNodeId mCheckInNodeId;
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ClusterCommand.h:268]: These members need to be documented in terms of how they are expected to be used.  Right now we are leaking them across command invocations, which is very much not good.

File: src/app/icd/client/DefaultICDClientStorage.cpp [hunk 3]
```diff
@@ -235,16 +235,22 @@ CHIP_ERROR DefaultICDClientStorage::Load(FabricIndex fabricIndex, std::vector<IC
...
         NodeId nodeId;
+        NodeId checkInNodeId;
         FabricIndex fabric;
         ReturnErrorOnFailure(reader.EnterContainer(ICDClientInfoType));
         // Peer Node ID
         ReturnErrorOnFailure(reader.Next(TLV::ContextTag(ClientInfoTag::kPeerNodeId)));
         ReturnErrorOnFailure(reader.Get(nodeId));
 
+        ReturnErrorOnFailure(reader.Next(TLV::ContextTag(ClientInfoTag::kCheckInNodeId)));
+        ReturnErrorOnFailure(reader.Get(checkInNodeId));
```
Comment (bzbarsky-apple) [inline src/app/icd/client/DefaultICDClientStorage.cpp:246]: This is a backwards-incompatible change to the storage format.  I assume that's OK?

File: src/controller/java/AndroidCheckInDelegate.cpp
Comment (bzbarsky-apple) [inline src/controller/java/AndroidCheckInDelegate.cpp:1]: I did not review the java bits here.

File: src/app/icd/client/DefaultICDClientStorage.cpp (hunk 3 above)
Comment (yunhanw-google) [reply src/app/icd/client/DefaultICDClientStorage.cpp:246]: yes, it should be ok considering 1.4 has not yet released and this is internal change.

File: examples/chip-tool/commands/clusters/ClusterCommand.h (hunk 2 above)
Comment (yunhanw-google) [reply examples/chip-tool/commands/clusters/ClusterCommand.h]: updated documentation, thanks

File: examples/chip-tool/commands/clusters/ClusterCommand.h (hunk 1 above)
Comment (yunhanw-google) [reply examples/chip-tool/commands/clusters/ClusterCommand.h:165]: removed, thanks

File: examples/chip-tool/commands/clusters/ClusterCommand.h [hunk 4]
```diff
@@ -264,10 +263,21 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
 private:
     chip::ClusterId mClusterId;
     chip::CommandId mCommandId;
+    // The scoped node ID to which RegisterClient and unregisterClient command will be sent
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ClusterCommand.h:266]:
```suggestion
    // The scoped node ID to which RegisterClient and UnregisterClient command will be sent.  Not set for other commands.
```

File: examples/chip-tool/commands/clusters/ClusterCommand.h [hunk 5]
```diff
@@ -264,10 +263,21 @@ class ClusterCommand : public InteractionModelCommands, public ModelCommand, pub
 private:
     chip::ClusterId mClusterId;
     chip::CommandId mCommandId;
+    // The scoped node ID to which RegisterClient and unregisterClient command will be sent 
     chip::ScopedNodeId mPeerNodeId;
+    // The scoped node ID to which a Check-In message will be sent
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/clusters/ClusterCommand.h:268]:
```suggestion
    // The scoped node ID to which a Check-In message will be sent.  Only set for the RegisterClient command.
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/java/src/matter/controller/ICDClientInfo.kt [hunk 6]
```diff
@@ -21,11 +21,12 @@ package matter.controller
 /** Class for holding ICD Client information. */
 data class ICDClientInfo(
   val peerNodeId: Long,
+  val checkInNodeId: Long
```
Comment (joonhaengHeo) [inline src/controller/java/src/matter/controller/ICDClientInfo.kt:24]: require to add ","

Comment (yunhanw-google) [reply src/controller/java/src/matter/controller/ICDClientInfo.kt]: fixed, thanks

**Pull request merged**